### PR TITLE
Fixes explosive lances not being explosive

### LINKED ISSUE #41372: Not so Explosive Lances
Issue reported from  Round ID: 96824 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 512

Recently tried using explosive lances to see if the syndie minibomb lance still worked. Turns out that explosive lances aren't exploding at all. Tried it with a minibomb, frag grenade, flashbang and a water/pot grenade. None of them worked aside from being a more expensive spear.

### PR DESCRIPTION
**Pull request by MrDoomBringer**
:cl: MrDoomBringer
fix: Explosive lances now explode again
/:cl:

Fixes #41372

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
gitmods are gonna have a field day with this PR

**Comment by Arianya** [comment]
small brain: keep explosive lances
big brain: remove explosive lances
galaxy brain: fix a "bug" and remove explosive lances "accidentally" :wink:

File: code/game/objects/items/twohanded.dm [hunk 1]
```diff
@@ -492,9 +492,10 @@
 
 /obj/item/twohanded/spear/afterattack(atom/movable/AM, mob/user, proximity)
 	. = ..()
+	
 	if(!proximity)
 		return
-	if(isopenturf(AM)) //So you can actually melee with it
+	if(!istype(src, /obj/item/twohanded/spear/explosive) && isopenturf(AM)) //So you can actually melee with it
 		return
```
Comment (ninjanomnom) [inline code/game/objects/items/twohanded.dm]: ...these 2 if/return statements are doing nothing

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/objects/items/twohanded.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/game/objects/items/twohanded.dm]: what did the openturf one do originally? i could never figure out

File: code/game/objects/items/twohanded.dm [hunk 2]
```diff
@@ -492,10 +492,9 @@
 
 /obj/item/twohanded/spear/afterattack(atom/movable/AM, mob/user, proximity)
```
Comment (SpaceManiac) [inline code/game/objects/items/twohanded.dm]: this entire override does nothing and can be removed

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Review by SpaceManiac — APPROVED** [review]

**Comment by subject217** [comment]
RIP

**Comment by Fikou** [comment]
bruh

**Pull request merged**